src/commands/createNewProject/javaSteps/JavaVersionStep.ts

```diff
@@ -4,29 +4,46 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzureWizardPromptStep, IAzureQuickPickItem } from "@microsoft/vscode-azext-utils";
+import { previewDescription } from "../../../constants";
 import { hasMinFuncCliVersion } from "../../../funcCoreTools/hasMinFuncCliVersion";
 import { localize } from "../../../localize";
 import { IJavaProjectWizardContext } from "./IJavaProjectWizardContext";
 
 export const java8: string = '8';
 export const java11: string = '11';
+export const java17: string = '17';
+
+const versionInfo: [string, string, string, string?][] = [
+    [java8, 'Java 8', '1.0.0'],
+    [java11, 'Java 11', '3.0.2630'],
+    [java17, 'Java 17', '4.0.0', previewDescription]
+];
 
 export class JavaVersionStep extends AzureWizardPromptStep<IJavaProjectWizardContext> {
+
     public static async setDefaultVersion(context: IJavaProjectWizardContext): Promise<void> {
         if (!await hasMinFuncCliVersion(context, '3.0.2630', context.version)) {
             context.javaVersion = java8;
         }
     }
 
     public async prompt(context: IJavaProjectWizardContext): Promise<void> {
-        const picks: IAzureQuickPickItem<string>[] = [
-            { label: 'Java 8', data: java8 },
-            { label: 'Java 11', data: java11 },
-        ];
+        const picks: IAzureQuickPickItem<string>[] = await this.getPicks(context);
         const placeHolder: string = localize('selectJavaVersion', 'Select a version of Java');
         context.javaVersion = (await context.ui.showQuickPick(picks, { placeHolder })).data;
     }
 
+    // todo: get runtime from Get Function App Stacks API and validate local java version
```

Comment:
We don't typically allow code to be checked in with a TODO comment in it. Is this not possible to be done today?

Reply:
todo validation has added